commands/utils/git.go

```diff
@@ -215,6 +231,48 @@ func (gm *GitManager) IsClean() (bool, error) {
 	return status.IsClean(), nil
 }
 
+func (gm *GitManager) GenerateCommitTitle(impactedPackage string, version string) string {
+	format := gm.customFormats.commitTitleFormat
+	if format == "" {
+		format = CommitTitleFormat
+	}
+	return formatStringWithPlaceHolders(format, impactedPackage, version, true)
+}
+
+func formatStringWithPlaceHolders(format, impactedPackage, fixVersion string, allowSpaces bool) string {
+	str := strings.Replace(strings.Replace(format, PackagePlaceHolder, impactedPackage, 1), FixVersionPlaceHolder, fixVersion, 1)
+	if allowSpaces {
+		return str
+	} else {
+		return strings.ReplaceAll(str, " ", "_")
+	}
+}
+
+func (gm *GitManager) GenerateFixBranchName(branch string, impactedPackage string, version string) (string, error) {
+	uniqueString, err := Md5Hash("frogbot", branch, impactedPackage, version)
+	if err != nil {
+		return "", err
+	}
+	// Package names in Maven usually contain colons, which are not allowed in a branch name
+	fixedPackageName := strings.ReplaceAll(impactedPackage, ":", "_")
+	branchFormat := gm.customFormats.branchNameFormat
+	if branchFormat == "" {
+		branchFormat = NewBranchesFormat
+	}
+	// Unique string is not optional
+	branchName := formatStringWithPlaceHolders(branchFormat, fixedPackageName, version, false)
+	return branchName + "-" + uniqueString, nil
```

Comment:
This is wrong. We cannot force the hash to be at the end. It should be placed according to the format set by the user.